packages/eslint-config-nimble/typescript.js

```diff
@@ -32,7 +32,16 @@ module.exports = {
         'require-await': 'off',
         '@typescript-eslint/require-await': 'error',
         'no-return-await': 'off',
-        '@typescript-eslint/return-await': 'error'
+        '@typescript-eslint/return-await': 'error',
+
+        // Increasing the value of minProperties (default is 6?) so that eslint doesn't
+        // introduce line breaks where prettier doesn't. If eslint introduces line breaks,
+        // they will be unix-style, which will cause pointless diffs in git.
+        'object-curly-newline': ['error', {
```

Comment:
This is a general eslint config rule, not a typescript-eslint specific rule (notice the name does not begin with `@typescript-eslint`) can you move this to the JS rule set?

Reply:
Oh there actually is a typescript eslint rule override: https://typescript-eslint.io/rules/object-curly-spacing/

1. So the current rule needs to go in the JS ruleset, and in the TS rules the JS needs to be disabled and the ts rule enabled, i.e.
  
   js rules:
    ```
    'object-curly-newline': ['error', {
        ...blah
    }]
    ```
  
    ts rules:
    ```
    'object-curly-newline': 'off',
    '@typescript-eslint/object-curly-newline': ['error', {
        ...blah
    }]
    ```
2. Can you create an issue in the ni style guide to enable the typescript-eslint object-curly-newline override rule

Reply:
Created https://github.com/ni/javascript-styleguide/issues/147. I wasn't sure if we want to change the `minProperties` value for the JS and TS versions of that rule, so I didn't include that in the issue. My guess is that since our motivation for increasing it is specific to using `prettier-eslint` (and not just `eslint` by itself), we want to leave `minProperties` at 6 in the styleguide.

Reply:
Ahh, saw the issue close, google search did me dirty (and not reading close enough) lol

Reply:
Whoops. Noticed a little too late that there _isn't_ a @typescript-eslint version of the rule. There is a typescript rule `object-curly-spacing`, but not `-newline`. I closed the issue that I had created.

Since the typescript config doesn't inherit from the javascript one (should it?), I have to configure the rule in both the ts and js configs.

Reply:
configuring in both is fine for now, there are challenges making them inherit not worth dealing with atm